File: OCH_Tao/ABC/A/249.js
const [A, B, C, D, E, F, X] = ((require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"))[0].split(" ")).map(x => parseInt(x, 10));
let Taka = 0;
let Ao = 0;
for (let i = 0; i < X; i++) {
  if (i % (A + C) < A) {
    Taka += B;
  }
  if (i % (D + F) < D) {
    Ao += E;
  }
}
if (Taka === Ao) {
  console.log("Draw");
} else if (Taka > Ao) {
  console.log("Takahashi");
} else {
  console.log("Aoki");
}

File: OCH_Tao/ABC/A/249.py
A,B,C,D,E,F,X = map(int,input().split())
sT = X//(A+C)
tT = sT*A
tT += X%(A+C) if X%(A+C)<A else A
sA = X//(D+F)
tA = sA*D
tA += X%(D+F) if X%(D+F)<D else D
if tT*B>tA*E:
  print("Takahashi")
elif tT*B<tA*E:
  print("Aoki")
else:
  print("Draw")